sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/HardwareSecurityModulesPrivateEndpointConnectionData.cs

```diff
@@ -0,0 +1,65 @@
+    {
+        /// <summary> Initializes a new instance of HardwareSecurityModulesPrivateEndpointConnectionData. </summary>
+        public HardwareSecurityModulesPrivateEndpointConnectionData()
+        {
+            GroupIds = new ChangeTrackingList<string>();
+        }
+
+        /// <summary> Initializes a new instance of HardwareSecurityModulesPrivateEndpointConnectionData. </summary>
+        /// <param name="id"> The id. </param>
+        /// <param name="name"> The name. </param>
+        /// <param name="resourceType"> The resourceType. </param>
+        /// <param name="systemData"> The systemData. </param>
+        /// <param name="etag"> Modified whenever there is a change in the state of private endpoint connection. </param>
+        /// <param name="privateEndpoint"> The private endpoint resource. </param>
+        /// <param name="connectionState"> A collection of information about the state of the connection between service consumer and provider. </param>
+        /// <param name="provisioningState"> The provisioning state of the private endpoint connection resource. </param>
+        /// <param name="groupIds"> The group ids for the private endpoint resource. </param>
+        internal HardwareSecurityModulesPrivateEndpointConnectionData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, ETag? etag, SubResource privateEndpoint, HardwareSecurityModulesPrivateLinkServiceConnectionState connectionState, HardwareSecurityModulesPrivateEndpointConnectionProvisioningState? provisioningState, IReadOnlyList<string> groupIds) : base(id, name, resourceType, systemData)
+        {
+            ETag = etag;
+            PrivateEndpoint = privateEndpoint;
+            ConnectionState = connectionState;
+            ProvisioningState = provisioningState;
+            GroupIds = groupIds;
+        }
+
+        /// <summary> Modified whenever there is a change in the state of private endpoint connection. </summary>
+        public ETag? ETag { get; set; }
+        /// <summary> The private endpoint resource. </summary>
+        internal SubResource PrivateEndpoint { get; set; }
+        /// <summary> Gets Id. </summary>
+        public ResourceIdentifier PrivateEndpointId
+        {
+            get => PrivateEndpoint is null ? default : PrivateEndpoint.Id;
+        }
+
+        /// <summary> A collection of information about the state of the connection between service consumer and provider. </summary>
+        public HardwareSecurityModulesPrivateLinkServiceConnectionState ConnectionState { get; set; }
+        /// <summary> The provisioning state of the private endpoint connection resource. </summary>
+        public HardwareSecurityModulesPrivateEndpointConnectionProvisioningState? ProvisioningState { get; }
+        /// <summary> The group ids for the private endpoint resource. </summary>
+        public IReadOnlyList<string> GroupIds { get; }
```

Comment:
What's the format of the element of this list?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).